packages/swingset-theme-hashicorp/src/components/app-wrapper.tsx

```diff
@@ -0,0 +1,26 @@
+'use client'
+import { SideNavigation } from './side-nav'
```

Comment:
Given that we have `'use client'` here and we're importing `side-nav`, `SideNavigation` will be rendered as a client component. This is not a necessarily a problem, but if your intent was to keep `SideNavigation` is a server component, we would need to adjust how it is imported and rendered. Given that we are passing in state values from `AppWrapper` into `SideNavigation` though, what you have here might be the most reasonable setup.

Reply:
Agreed
I wanted to avoid passing state to basically the entire tree but as you said there may not be much of a way around it